src/main/java/mate/academy/spring/boot/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package mate.academy.spring.boot.service.impl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.spring.boot.dto.BookDto;
+import mate.academy.spring.boot.dto.CreateBookRequestDto;
+import mate.academy.spring.boot.exception.EntityNotFoundException;
+import mate.academy.spring.boot.mapper.BookMapper;
+import mate.academy.spring.boot.model.Book;
+import mate.academy.spring.boot.repositry.BookRepository;
+import mate.academy.spring.boot.service.BookService;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto bookDto) {
+        Book book = bookMapper.createBookRequestDtoToBook(bookDto);
+        book = bookRepository.save(book);
+        return bookMapper.bookToBookDto(book);
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::bookToBookDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Book not found"));
```

Comment:
Use more informative exception message
```suggestion
                .orElseThrow(() -> new EntityNotFoundException(String.format("Book with id: %s not found", id));
```